scripts/Dockerfile

```diff
@@ -28,11 +29,9 @@ RUN uv venv $VIRTUAL_ENV
 RUN #!/bin/bash && source /home/pybamm/venv/bin/activate;
 ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
-RUN uv pip install --upgrade setuptools wheel wget cmake
-
+RUN uv pip install --upgrade scikit-build-core==0.10.3 wget cmake casadi pybind11
```

Comment:
Yes, scikit-build-core only provides experimental support for editable-rebuild installs.
1) To avoid unwanted breaking changes
2) Adding bounds helps with reproducibility in some aspects if someone were to build PyBaMM's older releases from source some time down the line.

Hence the necessity to pin scikit-build-core. For more info, have a look at the [documentation](https://scikit-build-core.readthedocs.io/en/stable/configuration.html#editable-installs).

Reply:
Oh, I will update it. Thanks for pointing out!